src/store/group-management/saga.ts

```diff
@@ -93,3 +100,30 @@ export function* roomMembersSelected(action) {
     yield put(setIsAddingMembers(false));
   }
 }
+
+export function* editConversationNameAndIcon(action) {
+  const { roomId, name, image } = action.payload;
+
+  yield put(setEditConversationState(EditConversationState.INPROGRESS));
+  try {
+    let imageUrl = '';
+    if (image) {
+      try {
+        const uploadResult = yield call(uploadImage, image);
+        imageUrl = uploadResult.url;
+      } catch (error) {
+        yield put(setEditConversationErrors({ image: 'Failed to upload image, please try again...' }));
+        return;
+      }
+    }
+
+    const chatClient: Chat = yield call(chat.get);
+    yield call([chatClient, chatClient.editRoomNameAndIcon], roomId, name, imageUrl);
+    yield put(setEditConversationState(EditConversationState.SUCCESS));
```

Comment:
Should we also clear the errors on success?

Reply:
Yes we should. Will add. (doubt) But i guess..we'll never reach this stage If there is ANY error (image/general)?

Reply:
Right but, if I tried to save and there was an error...and then I try to save again and it succeeds, won't the errors still be there? Maybe I'm mistaken.